src/api/c/memoryapi.hpp

```diff
@@ -0,0 +1,31 @@
+/*******************************************************
+ * Copyright (c) 2019, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#include <events.hpp>
+#include <af/event.h>
+#include <af/memory.h>
+
+struct BufferInfo {
+    void *ptr;
+    af_event event;
+};
+
+BufferInfo &getBufferInfo(const af_buffer_info pair);
+
+af_buffer_info getHandle(BufferInfo &pairHandle);
+
+detail::Event &getEventFromBufferInfoHandle(const af_buffer_info handle);
+
+struct MemoryManager {};
```

Comment:
May be I missed noticing where this is defined completely. Is this `MemoryManager` actually an empty struct or forward declaration ?

Reply:
This isn't complete yet - I've forgotten to add a todo here. Sorry. You can ignore this for now. I wanted to get comments on the new hierarchy/creation of `BackendMemoryClient` (or whatever we decide to call it).